spec/models/room_spec.rb

```diff
@@ -54,6 +54,111 @@
     end
   end
 
+  describe 'private methods' do
+    describe '#generate_code' do
+      before do
+        allow_any_instance_of(described_class).to receive(:generate_code).and_call_original
+      end
+
+      it 'calls SecureRandom#alphanumeric(6) and downcase its returned value' do
+        allow(SecureRandom).to receive(:alphanumeric).and_return('TEST')
+        expect(SecureRandom).to receive(:alphanumeric).with(6)
+        expect(room.generate_code).to eq('test')
+      end
+    end
+  end
+
+  describe '#viewer_access_code' do
+    it 'calls #get_setting with name "glViewerAccessCode" and returns the value' do
+      allow_any_instance_of(described_class).to receive(:get_setting).and_return(instance_double(RoomMeetingOption, value: 'CODE'))
```

Comment:
Can we stay consistent with the other tests and not use `instance_double`?

Reply:
I'm afraid that this is different,
It's the same case such as [THIS](https://github.com/bigbluebutton/greenlight/blob/v3/spec/models/room_spec.rb#L36) and [THIS](https://github.com/bigbluebutton/greenlight/blob/v3/spec/models/room_spec.rb#L43).
We're expecting an object with a method `value`.

Reply:
More is listed in #3626